posthog/special_migrations/runner.py

```diff
@@ -0,0 +1,114 @@
+from asyncio.tasks import sleep
+from datetime import datetime, timedelta
+
+from posthog.celery import app
+from posthog.models.special_migration import MigrationStatus, SpecialMigration
+from posthog.models.utils import UUIDT
+from posthog.special_migrations.manager import ALL_SPECIAL_MIGRATIONS
+
+
+# select for update?
+def start_special_migration(migration_name):
+    migration_instance = SpecialMigration.objects.get(name=migration_name, status=MigrationStatus.NotStarted)
+    if not migration_instance:
+        return False
+
+    migration_definition = ALL_SPECIAL_MIGRATIONS[migration_name].Migration
+
+    for service_version_requirement in migration_definition.service_version_requirements:
+        [in_range, version] = service_version_requirement.is_service_in_accepted_version()
+        if not in_range:
+            process_error(
+                migration_instance,
+                f"Service {service_version_requirement.service} is in version {version}. Expected range: {str(service_version_requirement.supported_version)}.",
+            )
```

Comment:
Potentially would be good to look at service requirements together with Posthog version + same handling. Imagine that we have a migration that requires <v5 and we already upgraded past it. We might want to have the restrictions listed in the UI & potentially even run the pre-check automatically on page load for the immediately next migration that the user can run.

Reply:
Actually I built #7283 with the intention of using it for posthog as well and my first change here was adding `posthog` to `ServiceVersionRequirement` supported services. 

However, they are fundamentally different, hence I changed it.

For e.g. ClickHouse version reqs, I don't want to run the _migration_ if CH is in a non-supported version. For PostHog version reqs, when above the max, I don't want to run _PostHog_ if the migration isn't done. Hence these are separate.